components/dashboard/src/teams/TeamUsageBasedBilling.tsx

```diff
@@ -0,0 +1,240 @@
+            const setupIntentId = params.get("setup_intent")!;
+            window.history.replaceState({}, "", window.location.pathname);
+            await getGitpodService().server.subscribeTeamToStripe(team.id, setupIntentId);
+            const pendingCustomer = { pendingSince: Date.now() };
+            setPendingStripeCustomer(pendingCustomer);
+            window.localStorage.setItem(`pendingStripeCustomerForTeam${team.id}`, JSON.stringify(pendingCustomer));
+        })();
+    }, [location.search, team]);
+
+    useEffect(() => {
+        setPendingStripeCustomer(undefined);
+        if (!team) {
+            return;
+        }
+        try {
+            const pendingStripeCustomer = window.localStorage.getItem(`pendingStripeCustomerForTeam${team.id}`);
+            if (!pendingStripeCustomer) {
+                return;
+            }
+            const pending = JSON.parse(pendingStripeCustomer);
+            setPendingStripeCustomer(pending);
+        } catch (error) {
+            console.error("Could not load pending stripe customer", team.id, error);
+        }
+    }, [team]);
+
+    useEffect(() => {
+        if (!pendingStripeCustomer || !team) {
+            return;
+        }
+        if (!!stripeCustomerId) {
+            // The upgrade was successful!
+            window.localStorage.removeItem(`pendingStripeCustomerForTeam${team.id}`);
+            clearTimeout(pollStripeCustomerTimeout!);
+            setPendingStripeCustomer(undefined);
+            return;
+        }
+        if (pendingStripeCustomer.pendingSince + 1000 * 60 * 5 < Date.now()) {
+            // Pending Stripe customer expires after 5 minutes
+            window.localStorage.removeItem(`pendingStripeCustomerForTeam${team.id}`);
+            clearTimeout(pollStripeCustomerTimeout!);
+            setPendingStripeCustomer(undefined);
+            return;
+        }
+        if (!pollStripeCustomerTimeout) {
+            // Refresh Stripe customer in 5 seconds in order to poll for upgrade confirmation
+            const timeout = setTimeout(async () => {
+                const customerId = await getGitpodService().server.getTeamStripeCustomerId(team.id);
+                setStripeCustomerId(customerId);
+                setPollStripeCustomerTimeout(undefined);
+            }, 5000);
+            setPollStripeCustomerTimeout(timeout);
+        }
+        return function cleanup() {
+            clearTimeout(pollStripeCustomerTimeout!);
+        };
+    }, [pendingStripeCustomer, pollStripeCustomerTimeout, stripeCustomerId, team]);
+
+    if (!showUsageBasedUI) {
+        return <></>;
+    }
+
+    return (
+        <div className="mb-16">
+            <h3>Usage-Based Billing</h3>
+            <h2 className="text-gray-500">Manage usage-based billing, spending limit, and payment method.</h2>
+            <div className="max-w-xl">
+                <div className="mt-4 h-32 p-4 pt-6 pl-6 flex flex-col rounded-xl bg-gray-100 dark:bg-gray-800">
```

Comment:
**nitpick**: Minor spacing fix to match the design specs.

```suggestion
                <div className="mt-4 h-32 p-4 pt-4 pl-4 flex flex-col rounded-xl bg-gray-100 dark:bg-gray-800">
```

Reply:
Ah, you're right. I knew I should be more careful with sharing sandbox design files and not final design specs that include only designs for one issue. Sorry for the confusion, @jankeromnes!

You probably checked out a previous iteration[[1](https://www.figma.com/file/8UO0eql1UPmI1B0zgf5Kcu/Plans-(Sandbox)?node-id=1021%3A3469)], not the latest in progress below[[2](https://www.figma.com/file/8UO0eql1UPmI1B0zgf5Kcu/Plans-(Sandbox)?node-id=1030%3A3958)].

| Early iteration | Last iteration |
|-|-|
| <img width="1440" alt="Team Plans" src="https://user-images.githubusercontent.com/120486/172441135-e6070183-df8b-4680-b23c-cd31e9a0ea9c.png"> | <img width="1440" alt="Team Plans-1" src="https://user-images.githubusercontent.com/120486/172441070-f4925c23-8e3d-473c-a0e3-620b578e1e00.png"> |